compiler.go

```diff
@@ -634,6 +643,40 @@ func (c *Compiler) EnableFileImport(enable bool) {
 	c.allowFileImport = enable
 }
 
+// SetImportDir sets the initial import directory path for file imports.
+func (c *Compiler) SetImportDir(dir string) {
+	c.importDir = dir
+}
+
+func (c *Compiler) resolveModuleLink(node parser.Node,
+	modulePath string, level int) (string, error) {
+	if level == 2 {
+		return "", c.errorf(node,
+			"module symbolic link max level is reached with \"%s\"", modulePath)
+	}
+	if info, err := os.Lstat(modulePath); err != nil {
+		return "", c.errorf(node, "module file lstat error: %s in \"%s\"",
+			err.Error(), modulePath)
+	} else if info != nil && info.Mode()&os.ModeSymlink != 0 {
```

Comment:
are we following symlinks so we can determine an absolute path of the target module?

Reply:
Yes, I could not resist to implement a solution for symlinks but not a must I guess for now.

Reply:
Yeah, if it's only for cyclic dependency detection, let's remove it for now. We can open a separate PR for symlink later.

Reply:
:+1: